geode-core/src/main/java/org/apache/geode/internal/tcp/MsgStreamer.java

```diff
@@ -129,7 +130,8 @@ public ConnectExceptions getConnectExceptions() {
     this.stats = stats;
     this.msg = msg;
     this.cons = cons;
-    this.buffer = bufferPool.acquireDirectSenderBuffer(sendBufferSize);
+    int bufferSize = Math.min(sendBufferSize, Connection.MAX_MSG_SIZE);
+    this.buffer = bufferPool.acquireDirectSenderBuffer(bufferSize);
```

Comment:
since this class is in the same package as `Connection` this seems like appropriate coupling ✓